Utilise react-helmet-async instead of react-helmet

jamessspanggg opened the pull request:
Fixes #856 

Solution taken from: nfl/react-helmet#548

Acc No: c54c04774efaae20746fd3f29cdd619fea512e119bc1082b863572b2e8844104

thesanjeevsharma commented:
@jamessspanggg I checked your solution solves the issue but it also introduced another bug.

When I go to any route for the first time, the title changes. But after that no matter how many routes I change, the title doesn't change at all. It always shows the title of the first page I visited.

Just try navigating to different pages and notice title.

jamessspanggg commented:
> @jamessspanggg I checked your solution, it solves the issue but it also introduced another bug.
> 
> When I go to any route for the first time, the title changes. But after that no matter how many routes I change, the title doesn't change at all. It always shows the title of the first page I visited.
> 
> Just try navigating to different pages and notice title.

My bad, fixed it. I moved the Provider out to the App component.
